pyscf/pbc/scf/_response_functions.py

```diff
@@ -189,6 +189,26 @@ def _gen_ghf_response(mf, mo_coeff=None, mo_occ=None,
     '''
     raise NotImplementedError
 
+def _get_jk_kshift(mf, dm_kpts, hermi, kpts, kshift, with_j=True, with_k=True):
+    from pyscf.pbc.df.df_jk import get_j_kpts_kshift, get_k_kpts_kshift
+    vj = vk = None
+    if with_j:
+        vj = get_j_kpts_kshift(mf.with_df, dm_kpts, kshift, hermi=hermi, kpts=kpts)
```

Comment:
`mf.get_jk` may call the `mf.rsjk.get_jk` method. Check if mf.rsjk is initialized as the get_jk code implemented in `scf/khf.py` . If rsjk object available, an assertion to `kshift=0` may be needed I guess.

Reply:
The current code checks `isinstance(mf.with_df, df.DF)`, which evaluates to `True` for `rsjk` since it uses GDF for `vnuc` build. As a result, the code will automatically cache j3c if roots with non-zero `kshift` are requested. What do you think this behavior (compared to an `assert` error)? If so I can add a warning for `rsjk`.

Reply:
I think an error can be raised here. We just need to ensure that users do not get incorrect results. It is not a problem if this combination is not supported.

Reply:
No problem. I added a `NotImplementedError` for rsjk in both `pbc/tdscf/krhf.py` and `pbc/scf/_response_functions.py`. Please let me know if it looks good to you.